src/state/reducers/dataset/DatasetsReducer.js

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import { createReducer } from '@reduxjs/toolkit';
+import { ResourceUtils } from '@cosmotech/core';
+import { STATUSES } from '../../commons/Constants';
+import { DATASET_ACTIONS_KEY } from '../../commons/DatasetConstants';
+
+export const datasetInitialState = {
+  list: {
+    data: [],
+    status: STATUSES.IDLE,
+  },
+  selectedDatasetIndex: null,
+};
+
+export const datasetsReducer = createReducer(datasetInitialState, (builder) => {
+  builder
+    .addCase(DATASET_ACTIONS_KEY.GET_ALL_DATASETS, (state, action) => {
+      state.list.status = STATUSES.LOADING;
+      state.selectedDatasetIndex = null;
+    })
+    .addCase(DATASET_ACTIONS_KEY.SET_ALL_DATASETS, (state, action) => {
+      state.list.data = action.list;
+      state.list.status = action.status;
+      // state.selectedDatasetIndex = action.list?.length > 0 ? null : action.selectedDatasetIndex;
```

Comment:
Remaining commented line to remove :slightly_smiling_face:

Reply:
nice catch, thanks ! (but :woman_facepalming: )